popgen/MCMC_Selection.go

```diff
@@ -0,0 +1,103 @@
+package popgen
+
+import (
+	"github.com/vertgenlab/gonomics/common"
+	"github.com/vertgenlab/gonomics/numbers"
+	"math/rand"
+	//DEBUG: "fmt"
+)
+
+type Theta struct {
+	alpha       []float64
+	mu          float64
+	sigma       float64
+	probability float64
+}
+
+func Metropolis_Accept(old Theta, thetaPrime Theta, data AFS, logSpace bool) bool {
+	yRand := rand.Float64()
+	var pAccept float64
+	pAccept = common.MinFloat64(1.0, Bayes_Ratio(old, thetaPrime, data, logSpace)*Hastings_Ratio(old, thetaPrime))
+	//DEBUG: fmt.Printf("Likelihood ratio: %f\n", pAccept)
+	return pAccept > yRand
+}
+
+func Hastings_Ratio(tOld Theta, tNew Theta) float64 {
+	var newGivenOld, oldGivenNew float64
+
+	newGivenOld = numbers.NormalDist(tNew.mu, tOld.mu, tOld.sigma) //* numbers.GammaDist(tNew.sigma, tOld.sigma*tOld.sigma, tOld.sigma)
+	oldGivenNew = numbers.NormalDist(tOld.mu, tNew.mu, tNew.sigma) //* numbers.GammaDist(tOld.sigma, tNew.sigma*tNew.sigma, tNew.sigma)
+
+	/*
+		for i := 0; i < len(tOld.alpha); i++ {
+			newGivenOld = newGivenOld * numbers.NormalDist(tNew.alpha[i], tNew.mu, tNew.sigma)
+			oldGivenNew = oldGivenNew * numbers.NormalDist(tOld.alpha[i], tOld.mu, tOld.sigma)
+		}*/
+	return oldGivenNew / newGivenOld
+}
+
+func Bayes_Ratio(old Theta, thetaPrime Theta, data AFS, logSpace bool) float64 {
+	if logSpace {
+		return AFSLogLikelihood(data, old.alpha) * old.probability / (AFSLogLikelihood(data, thetaPrime.alpha) * thetaPrime.probability)
+	}
+	return AFSLikelihood(data, old.alpha) * old.probability / (AFSLikelihood(data, thetaPrime.alpha) * thetaPrime.probability)
```

Comment:
I think that this should be "prime / old" instead of "old / prime".  The accept probability should go higher when the proposal has a higher likelihood.